sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs

```diff
@@ -123,7 +127,12 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
                     }
                 }
 
-                AzureCoreEventSource.Singleton.RequestRetrying(message.Request.ClientRequestId, attempt);
+                string? retryReason = null;
```

Comment:
Looking at it again, let's skip logging the reason and instead log the exception if it happens in LoggingPolicy.

The main motivation for logging a reason was to be able to know if there was an exception or not and we already logging details of the request in the LoggingPolicy.

Reply:
We are already catching [lastException ](https://github.com/Azure/azure-sdk-for-net/blob/3f8f8c3c939b873a2e65d6f92287e9931945bb6b/sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs#L73)here, do you want to log this in [RequestRetrying](https://github.com/Azure/azure-sdk-for-net/blob/3f8f8c3c939b873a2e65d6f92287e9931945bb6b/sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs#L135) instead of logging the reason?

Reply:
I was thinking about logging the exception in logging policy in this area: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/Internal/LoggingPolicy.cs#L71-L78


This would allow us to see the exception when retries happen.

Reply:
Don't we wanna log the exception for each retry?

https://github.com/Azure/azure-sdk-for-net/issues/9621#issue-554897043